Dockerfile

```diff
@@ -41,7 +41,12 @@ RUN \
   apt-get -qq update && \
   apt-get -qq install apt-utils 2>/dev/null && \
   apt-get -qq dist-upgrade && \
+  echo "applying patch for https://github.com/docker-mailserver/docker-mailserver/issues/2023" && \
+  mv /bin/hostname{,.bak} && \
+  echo "echo docker.mail.server" > /bin/hostname && \
+  chmod +x /bin/hostname && \
   apt-get -qq install postfix && \
+  mv /bin/hostname{.bak,} && \
```

Comment:
Personal preference, I'd rather we split the bounds of this patch with Postfix install into a separate `RUN`.

That's not required for this PR, but would be better for maintenance imo when the scope of relevant lines is clearer, layers is not a concern if we adopt multi-stage build.

The whole `RUN` chaining is primarily to minimize layers fattening an image with redundant data, but that's no longer a concern to worry about with multi-stage builds.

Reply:
I agree, but do `update` and `apt-utils` need to install first? Or can postfix be installed above them all?

Reply:
I vaguely remember looking into this and I think it may have been necessary. I used a Debian base image at the time and shelled into it, then ran the commands to see what would work.

I also see that I detailed this same workaround as one of the alternatives we could use if fixing it within the Dockerfile :sweat_smile:

Reply:
No worries, you just had me confused for a second :)

Reply:
Wait a sec, was that a reply to this thread, or https://github.com/docker-mailserver/docker-mailserver/pull/2468#discussion_r820312198 ?

Reply:
It was a reply regarding `update` and `apt-utils`, I glanced through my comments on the reference issue to see if I detailed that to help answer you here, and noticed hidden away I had detailed this workaround (just with a different hostname value).